src/ng/directive/ngOptions.js

```diff
@@ -643,7 +643,7 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
                 (current === emptyOption_ ||
                 current === unknownOption_ ||
                 current.nodeType === NODE_TYPE_COMMENT ||
-                current.value === '')) {
+                (current.nodeName == 'OPTION' && current.value === ''))) {
```

Comment:
Nitpick1: Use `===` instead of `==`.
Nitpick2: Might be better to use `nodeName_(current) === 'option'` for consistency.